core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartUpscaleGetStartedInstancesHandler.java

```diff
@@ -0,0 +1,130 @@
+package com.sequenceiq.cloudbreak.reactor.handler.cluster;
+
+import static com.sequenceiq.cloudbreak.cloud.model.HostName.hostName;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartUpscaleGetStartedInstancesRequest;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartUpscaleGetStartedInstancesResult;
+import com.sequenceiq.cloudbreak.cloud.handler.CloudPlatformEventHandler;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudInstance;
+import com.sequenceiq.cloudbreak.cloud.model.CloudVmInstanceStatus;
+import com.sequenceiq.cloudbreak.cloud.model.InstanceStatus;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterStatusService;
+import com.sequenceiq.cloudbreak.cluster.status.ExtendedHostStatuses;
+import com.sequenceiq.cloudbreak.converter.CloudInstanceIdToInstanceMetaDataConverter;
+import com.sequenceiq.cloudbreak.converter.spi.InstanceMetaDataToCloudInstanceConverter;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceMetaData;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.stack.RuntimeVersionService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopStartUpscaleGetStartedInstancesHandler implements CloudPlatformEventHandler<StopStartUpscaleGetStartedInstancesRequest> {
```

Comment:
Maybe rename to be more indicative of this step being a best effort error handling step. GetStartedInstances is to generic a name. Something along the lines of GetStartRecoveryCandidates / GetStopRecoveryCandidates.